internal/monitors/kubernetes/gen-metrics.py

```diff
@@ -0,0 +1,36 @@
+"""Creates metrics yaml section for openshift monitor"""
+import yaml
+import sys
+
+RESOURCES = {
+    "cpu": "number",
+    "memory": "amount",
+    "pods": "number",
+    "services": "number",
+    "persistentvolumeclaims": "number",
+    "services.nodeports": "number",
+    "services.loadbalancers": "number",
+}
+
+METRICS = []
+
+
+def build_metrics(metricPrefix, end):
+    for resource, typ in RESOURCES.items():
+        for category in ("hard", "used"):
+            if category == "hard":
+                description = f"Hard limit for {typ} of {resource} {end}"
+            else:
+                description = f"Consumed {typ} of {resource} {end}"
+
+            yield {
+                "description": description,
+                "included": False,
+                "name": f"{metricPrefix}.{resource}.{category}",
+                "type": "gauge",
+            }
+
+METRICS += build_metrics("openshift.clusterquota", "across all namespaces")
+METRICS += build_metrics("openshift.appliedclusterquota", "by namespace")
+
+yaml.safe_dump({"metrics": METRICS}, sys.stdout, encoding='utf-8', default_flow_style=False)
```

Comment:
Not really liking this.  If you really want to map the openshift monitor a superset of kubernetes-cluster, can't you just add another field to the metadata.yaml like `extendMetricsFrom: kubernetes-cluster` or something and have the monitor-code-gen thing read that and do the right thing?

Reply:
This script just generates the openshift-specific metrics. Yeah I think the only thing to do is do extendMetricsFrom. The downside being any consumer of the yaml files (like the tests) has to know about it and do populate the extended metrics.

Reply:
Hmm, would it be possible to use the `#from` thing? I'm not entirely sure how to reference the other metrics in the value. But one idea...

```yaml
  - {"#from": "Monitors['kubernetes-cluster']['metrics']", default: [], flatten: true}
  - description: Consumed number of pods across all namespaces
    included: false
    name: openshift.clusterquota.pods.used
    type: gauge
...
```

Reply:
not very easily, that is a pretty complicated mechanism and it's tied in pretty closely to the agent config loader.